presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,228 @@
+{
+    private static final Logger LOG = Logger.get(ElasticsearchQueryBuilder.class);
+
+    private final Duration scrollTimeout;
+    private final int scrollSize;
+    private final TransportClient client;
+    private final int shard;
+    private final TupleDomain<ColumnHandle> tupleDomain;
+    private final List<ElasticsearchColumnHandle> columns;
+    private final String index;
+    private final String type;
+
+    public ElasticsearchQueryBuilder(List<ElasticsearchColumnHandle> columnHandles, ElasticsearchConnectorConfig config, ElasticsearchSplit split)
+    {
+        columns = requireNonNull(columnHandles, "columnHandles is null");
+        requireNonNull(config, "config is null");
+        requireNonNull(split, "split is null");
+        ElasticsearchTableDescription table = split.getTable();
+        tupleDomain = split.getTupleDomain();
+        index = split.getIndex();
+        type = table.getType();
+        shard = split.getShard();
+        try {
+            Settings settings = Settings.builder()
+                    .put("client.transport.ignore_cluster_name", true)
+                    .build();
+            client = new PreBuiltTransportClient(settings)
+                    .addTransportAddress(new TransportAddress(InetAddress.getByName(split.getSearchNode()), split.getPort()));
+        }
+        catch (IOException e) {
+            throw new PrestoException(ELASTIC_SEARCH_CONNECTION_ERROR, format("Error connecting to Elasticsearch SearchNode: %s with port: %d", split.getSearchNode(), split.getPort()), e);
+        }
+        scrollTimeout = config.getScrollTimeout();
+        scrollSize = config.getScrollSize();
+    }
+
+    public void close()
+    {
+        client.close();
+    }
+
+    public SearchRequestBuilder buildScrollSearchRequest()
+    {
+        String indices = index != null && !index.isEmpty() ? index : "_all";
+        SearchRequestBuilder searchRequestBuilder = client.prepareSearch(indices)
+                .setTypes(type)
+                .setSearchType(SearchType.QUERY_THEN_FETCH)
+                .setScroll(new TimeValue(scrollTimeout.toMillis()))
+                .setQuery(buildSearchQuery())
+                .setPreference("_shards:" + shard)
+                .setSize(scrollSize);
+        LOG.debug("ElasticSearch Request: " + searchRequestBuilder.toString());
+        return searchRequestBuilder;
+    }
+
+    public SearchScrollRequestBuilder prepareSearchScroll(String scrollId)
+    {
+        return client.prepareSearchScroll(scrollId).setScroll(new TimeValue(scrollTimeout.toMillis()));
+    }
+
+    private QueryBuilder buildSearchQuery()
+    {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        for (ElasticsearchColumnHandle column : columns) {
+            BoolQueryBuilder columnQueryBuilder = new BoolQueryBuilder();
+            Type type = column.getColumnType();
+            tupleDomain.getDomains().ifPresent((domains) -> {
```

Comment:
`(domains)` -> `domains`

Reply:
Also personally I prefer not to use `ifPresent` if the `Consumer` passed to it is long. I prefer:

```
if (tupleDomain.getDomains().isPresent()) {
    Domain domain = tupleDomain.getDomains().get().get(column);
    if (domain != null) {
        columnQueryBuilder.should(buildPredicate(column.getColumnJsonPath(), domain, type));
    }

}
```